agent/src/workload/workload_control_loop.rs

```diff
@@ -97,8 +97,22 @@ impl WorkloadControlLoop {
                         )
                         .await;
 
+                        let restart_policy = &control_loop_state.workload_spec.restart_policy;
+
                     // [impl->swdd~workload-control-loop-handles-workload-restarts~1]
-                    control_loop_state = Self::handle_restart_on_received_workload_state(control_loop_state, new_workload_state).await;
+                    if Self::is_restart_required(&control_loop_state.workload_spec, &new_workload_state) {
+                        log::debug!(
+                            "Restart workload '{}' with restart policy '{}' caused by current execution state '{}'.",
+                            control_loop_state
+                                .workload_spec
+                                .instance_name
+                                .workload_name(),
+                            restart_policy,
+                            new_workload_state.execution_state
+                        );
```

Comment:
We could put the trace in the new `Self::restart_workload` function below to keep the code here clean

Reply:
Done, with shortening the log.